linter.py

```diff
@@ -74,13 +76,13 @@ def cmd(self):
         """
 
         result = self.base_cmd
+        settings = self.get_view_settings()
 
         if persist.get_syntax(self.view) in ['c', 'c improved']:
-            result += ' -x c '
+            result += ' -x c ' + settings.get('extra_cflags', '') + ' '
         elif persist.get_syntax(self.view) in ['c++', 'c++11']:
-            result += ' -x c++ '
+            result += ' -x c++ ' + settings.get('extra_cxxflags', '') + ' '
```

Comment:
If `apply_template` is used for `extra_flags` it should be used for the new flags too. 
That said, I see no point in it, so I would only use it for `include_dirs` and drop it for `extra_flags`.